### PR TITLE
Docs/contributor docs

### PR DESCRIPTION
**Pull request by abegong**
First pass on #216 

A couple questions:
* Should `CONTRIBUTING` be `CONTRIBUTING.md`
* Should we replicate this material in readthedocs?

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,60 @@
+
+
+Desired PR workflow
+Info on setup / running tests
+Conventions / style preferences.
```
Comment (louispotok) [inline CONTRIBUTING]: Remove the outline?

File: CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## How to contribute
+We're excited for contributions to Great Expectations. If you see places where the code or documentation could be improved, please get involved!
+
+Submitting your changes
+Once your changes and tests are ready to submit for review:
+
+1. Test your changes
+
+    Run the test suite to make sure that nothing is broken. See the the section on testing below for help running tests. (Hint: `pytest` from the great_expectations root.)
```
Comment (louispotok) [inline CONTRIBUTING:10]: If pytest is the recommended / supported test runner and discovery tool, should it also be in requirements-dev?

File: CONTRIBUTING (hunk 1 above)
Comment (abegong) [reply CONTRIBUTING]: :+1:

File: CONTRIBUTING (hunk 2 above)
Comment (abegong) [reply CONTRIBUTING:10]: Good catch

**Review by jpcampb2 — APPROVED** [review]
LGTM. Hopefully we'll quickly finish the changes to the testing framework and be able to pull our dirty laundry out of there, but I'm fine leaving it in as the current reality.

**Comment by abegong** [comment]
:+1:

**Comment by abegong** [comment]
I added pytest to `requirements-dev.txt`. We still need to switch travis to use pytest instead of unittest.

New issue created: #223.

**Pull request merged**